src/DotNetty.Buffers/ByteBufferUtil.cs

```diff
@@ -3,10 +3,229 @@
+        /// Returns a <a href="http://en.wikipedia.org/wiki/Hex_dump">hex dump</a>
+        /// of the specified buffer's sub-region.
+        /// </summary>
+        public static string HexDump(IByteBuffer buffer)
+        {
+            return HexDump(buffer, buffer.ReaderIndex, buffer.ReadableBytes);
+        }
+
+        /// <summary>
+        /// Returns a <a href="http://en.wikipedia.org/wiki/Hex_dump">hex dump</a>
+        /// of the specified buffer's sub-region.
+        /// </summary>
+        public static string HexDump(IByteBuffer buffer, int fromIndex, int length)
+        {
+            if (length < 0)
+            {
+                throw new ArgumentOutOfRangeException("length: " + length);
+            }
+            if (length == 0)
+            {
+                return "";
+            }
+            int endIndex = fromIndex + length;
+            char[] buf = new char[length << 1];
+
+            int srcIdx = fromIndex;
+            int dstIdx = 0;
+            for (; srcIdx < endIndex; srcIdx++, dstIdx += 2)
+            {
+                Array.Copy(
+                    HexdumpTable, buffer.GetByte(srcIdx) << 1,
+                    buf, dstIdx, 2);
+            }
+
+            return new string(buf);
+        }
+
+        /// <summary>
+        /// Returns a <a href="http://en.wikipedia.org/wiki/Hex_dump">hex dump</a>
+        /// of the specified buffer's sub-region.
+        /// </summary>
+        public static string HexDump(byte[] array)
+        {
+            return HexDump(array, 0, array.Length);
+        }
+
+        /// <summary>
+        /// Returns a <a href="http://en.wikipedia.org/wiki/Hex_dump">hex dump</a>
+        /// of the specified buffer's sub-region.
+        /// </summary>
+        public static string HexDump(byte[] array, int fromIndex, int length)
+        {
+            if (length < 0)
+            {
+                throw new ArgumentOutOfRangeException("length: " + length);
+            }
+            if (length == 0)
+            {
+                return "";
+            }
+
+            int endIndex = fromIndex + length;
+            char[] buf = new char[length << 1];
+
+            int srcIdx = fromIndex;
+            int dstIdx = 0;
+            for (; srcIdx < endIndex; srcIdx++, dstIdx += 2)
+            {
+                Array.Copy(HexdumpTable, (array[srcIdx] & 0xFF) << 1, buf, dstIdx, 2);
+            }
+
+            return new string(buf);
+        }
+
+        /// <summary>
+        /// Calculates the hash code of the specified buffer.  This method is
+        /// useful when implementing a new buffer type.
+        /// </summary>
+        public static int HashCode(IByteBuffer buffer)
+        {
+            int aLen = buffer.ReadableBytes;
+            int intCount = (int)((uint)aLen >> 2);
+            int byteCount = aLen & 3;
+
+            int hashCode = 1;
+            int arrayIndex = buffer.ReaderIndex;
+            if (buffer.Order == ByteOrder.BigEndian)
```

Comment:
We can check on reasoning with netty project, otherwise let's keep parity with netty code.